src/include/sof/audio/buffer.h

```diff
@@ -32,48 +32,43 @@
 
 struct comp_dev;
 
+/** \name Trace macros
+ *  @{
+ */
+
 /* buffer tracing */
-extern uintptr_t buffer_uuid_ptr;
-
-#define buf_cl_err(__e, ...) \
-	trace_error_with_ids(TRACE_CLASS_DEPRECATED, buffer_uuid_ptr, \
-			     -1, -1, __e, ##__VA_ARGS__)
-
-#define buf_cl_err_a(__e, ...) \
-	trace_error_atomic_with_ids(TRACE_CLASS_DEPRECATED, buffer_uuid_ptr, \
-				    -1, -1, __e, ##__VA_ARGS__)
-
-#define buf_cl_warn(__e, ...) \
-	trace_warn_with_ids(TRACE_CLASS_DEPRECATED, buffer_uuid_ptr, \
-			    -1, -1, __e, ##__VA_ARGS__)
-
-#define buf_cl_info(__e, ...) \
-	trace_event_with_ids(TRACE_CLASS_DEPRECATED, buffer_uuid_ptr, \
-			     -1, -1, __e, ##__VA_ARGS__)
-
-#define buf_cl_dbg(__e, ...) \
-	tracev_event_with_ids(TRACE_CLASS_DEPRECATED, buffer_uuid_ptr, \
-			      -1, -1, __e, ##__VA_ARGS__)
-
-#define buf_err(buf_ptr, __e, ...) \
-	trace_error_with_ids(TRACE_CLASS_DEPRECATED, buffer_uuid_ptr, \
-			     (buf_ptr)->pipeline_id, \
-			     (buf_ptr)->id, __e, ##__VA_ARGS__)
-
-#define buf_warn(buf_ptr, __e, ...) \
-	trace_warn_with_ids(TRACE_CLASS_DEPRECATED, buffer_uuid_ptr, \
-			    (buf_ptr)->pipeline_id, \
-			    (buf_ptr)->id, __e, ##__VA_ARGS__)
-
-#define buf_info(buf_ptr, __e, ...) \
-	trace_event_with_ids(TRACE_CLASS_DEPRECATED, buffer_uuid_ptr, \
-			     (buf_ptr)->pipeline_id, \
-			     (buf_ptr)->id, __e, ##__VA_ARGS__)
-
-#define buf_dbg(buf_ptr, __e, ...) \
-	tracev_event_with_ids(TRACE_CLASS_DEPRECATED, buffer_uuid_ptr, \
-			      (buf_ptr)->pipeline_id, \
-			      (buf_ptr)->id, __e, ##__VA_ARGS__)
+extern struct tr_ctx buffer_tr;
+
+/** \brief Retrieves trace context from the buffer */
+#define trace_buf_get_tr_ctx(buf_ptr) (&buffer_tr)
+
+/** \brief Retrieves id (pipe id) from the buffer */
+#define trace_buf_get_id(buf_ptr) ((buf_ptr)->pipeline_id)
+
```

Comment:
Would it be better to call this trace_buf_get_pipeline_id and the next call trace_buf_get_id?

Reply:
I'm also not a fan of this names, but I follow a convention.